pydepsi/io.py

```diff
@@ -0,0 +1,264 @@
+    az_resolutions = np.array([21.76, 21.89, 21.71])
+    np.array([2.63, 3.09, 3.51])
+    azimuth_resolution = az_resolutions[swath - 1]
+
+    # ++++ 20 - range_resolution
+    pattern = r"Total_range_band_width \(MHz\):" + SC_N_PATTERN
+    match = re.search(pattern, content)
+    range_resolution = SPPED_OF_LIGHT / (2 * float(match.group(1)) * 1e6)
+
+    # ++++ 21 - nBursts
+    burst_n = None
+
+    # ++++ 23 - steering_rate
+    pattern = r"Azimuth_steering_rate \(deg/s\):" + SC_N_PATTERN
+    match = re.search(pattern, content)
+    steering_rate = float(match.group(1)) * np.pi / 180
+
+    # ++++ 24 and 25 - azFmRateArray and dcPolyArray
+    # Are skipped because the io.datetimeToMJD function is missing
+
+    # ++++ 26 - pri
+    pattern = r"Pulse_Repetition_Frequency_raw_data\(TOPSAR\):" + SC_N_PATTERN
+    match = re.search(pattern, content)
+    pri = 1 / float(match.group(1))
+
+    # ++++ 27 - rank
+    # See Beta0 section
+
+    # ++++ 28 - chirp_rate
+
+    # ++++ 29 - n_azimuth
+    if crop_flag:
+        crop_file = "/".join(str(resfile).split("/")[0:-2]) + "/nlines_crp.txt"
+        with open(crop_file) as file:
+            content = file.readlines()
+            n_lines, first_line, last_line = (
+                int(content[0].strip()),
+                int(content[1].strip()),
+                int(content[2].strip()),
+            )
+
+    else:
+        # Extract first
+        pattern = r"First_line \(w.r.t. original_image\):" + SC_N_PATTERN
+        match = re.search(pattern, content)
+        first_line = int(match.group(1))
+        # Extract last
+        pattern = r"Last_line \(w.r.t. original_image\):" + SC_N_PATTERN
+        match = re.search(pattern, content)
+        last_line = int(match.group(1))
+        # difference
+        n_lines = last_line - first_line + 1
+
+    # ++++ 30 - n_range
+    if crop_flag:
+        crop_file = "/".join(str(resfile).split("/")[0:-2]) + "/npixels_crp.txt"
+        with open(crop_file) as file:
+            content = file.readlines()
+            n_pixels, first_pixel, last_pixel = (
+                int(content[0].strip()),
+                int(content[1].strip()),
+                int(content[2].strip()),
+            )
+    else:
+        # Extract first
+        pattern = r"First_pixel \(w.r.t. original_image\):" + SC_N_PATTERN
+        match = re.search(pattern, content)
+        first_pixel = int(match.group(1))
+        # Extract last
+        pattern = r"Last_pixel \(w.r.t. original_image\):" + SC_N_PATTERN
+        match = re.search(pattern, content)
+        last_pixel = int(match.group(1))
+        # difference
+        n_pixels = last_pixel - first_pixel + 1
+
+    # ----------------------------------------
+
+    # Fill the dictionary
+    datewise_metadata = {
+        "sat_id": sat_id,
+        "orbit": geometry,
+        "acq_date": acq_date,
+        "azimuth0time": azimuth0time,
+        "range0time": range0time / 2,
```

Comment:
Remove the /2 here, and move it to the actual reading/computation block, at line 61.

Reply:
Updated as suggested